fix: Refresh recipe section when clicking card tag chip

parumpum opened the pull request:
<!--
  This template provides some ideas of things to include in your PR description.

  To start, try providing a short summary of your changes in the Title above. We follow Conventional Commits syntax, please ensure your title is prefixed with one of:
  - `feat: `
  - `fix: `
  - `docs: `
  - `chore: `
  - `dev:`

  If a section of the PR template does not apply to this PR, then delete that section.

  PLEASE READ:
  -------------------------
  Mealie is moving to a regular, automatic release schedule. This means that all PRs should be in a
  stable state, ready for release. This includes:

  - Ensuring new tests have been added to cover new features, or to prevent regressions.
  - Work is fully complete and usable

 -->

## What type of PR is this?

<!--
  Delete any of the following that do not apply:
 -->

- bug


## What this PR does / why we need it:

On the recipe explore page, clicking a Tag inside a recipe card was rewriting the URL and not updating the page to reflect the Tag search.

This updates the Tag chip to append the tag query to the existing query, and trigger a refresh of the Recipe Section.

<!--
  What goal is this change working towards?
  Provide a bullet pointed summary of how each file was changed.
  Briefly explain any decisions you made with respect to the changes.
  Include anything here that you didn't include in *Release Notes*
  above, such as changes to CI or changes to internal methods.

  If there is a UI component to the change, please include before/after images.
-->

## Which issue(s) this PR fixes:

Fixes #4776 

<!--
If this PR fixes one of more issues, list them here.
One per line, like so:
Fixes #123
Fixes #39
-->

parumpum committed:
Refresh recipe section when clicking card tag

Kuchenpirat reviewed (changes requested):
Hey thanks for your contribution, always nice to see new people around helping out.

I had a quick look around and found a few bugs that need to be adressed:. 

1. The selected tag should be added to the tags filter below the search bar
2. Clicking on the mealie symbol on the top left should when already in search should clear the filter if already on the search window
3. Breaks functionality on pages that are not the main search page by just appending the filter instead of navigating to the main page and setting the filter.
3.1 Manage Data - Recipe (http://localhost:3000/group/data/recipes) - Break filter Button for Tools, Categories and Tags
3.2 Recipe Page - Breaks Buttons for Categories and Tags


Thats quite a bit, but i think they are mostly connected and can probably be adressed at once. If any questions arise feel free to reach out here or on the Discord and the dev team or the community will try to answer them.

parumpum committed:
Fix issues with query appending and nav

parumpum commented:
Addressed feedback. 

1. Clicking the tag properly checks the item in the dropdown below the search bar, refreshing the search.
2. Filter gets cleared when clicking Mealie on the top left
3. Not appending the query into the URL so addresses remaining issues.

Clicking the tag within /group/data/recipes will go to main page and search that tag. Can remove this piece if not desired behavior. Future work here would be to make the data table on /group/data/recipes filterable/searchable.

Kuchenpirat reviewed (changes requested):
Hey thanks for the quick turnaround.

1. & 2. Has been adressed by your changes and work as expected 👍

I still see issues with 3. 
- On the recipe page i cannot click any of the Categories and Tags of the recipe (which should redirect to the search page with the filter applied)
- On the Data Management page the behaviour is as expected for Tags but clicking on Categories and Tools does nothing. 

There are also some linting errors

parumpum committed:
Addressed feedback, use router instead of href

parumpum committed:
Fix some lint errors and mobile recipe card chips

parumpum committed:
Merge branch 'mealie-next' into fix/tag-filter

Kuchenpirat commented on frontend/components/Domain/Recipe/RecipePage/RecipePage.vue:
```diff
@@ -165,8 +165,8 @@ import {
 } from "~/composables/recipe-page/shared-state";
-import { NoUndefinedField } from "~/lib/api/types/non-generated";
-import { Recipe } from "~/lib/api/types/recipe";
+import { NoUndefinedField, RecipeOrganizer } from "~/lib/api/types/non-generated";
```
```suggestion
import { NoUndefinedField } from "~/lib/api/types/non-generated";
```

Kuchenpirat commented on frontend/components/Domain/Recipe/RecipeCardSection.vue:
```diff
@@ -141,6 +143,7 @@ import { useLazyRecipes } from "~/composables/recipes";
 import { Recipe } from "~/lib/api/types/recipe";
 import { useUserSortPreferences } from "~/composables/use-users/preferences";
 import { RecipeSearchQuery } from "~/lib/api/user/recipes/recipe";
+import { RecipeCategory } from "~/lib/api/types/household";
```
```suggestion
```

Kuchenpirat reviewed (commented):
This works now as expected, thanks for your work on this 😊

Just a few unused imports to remove and then we can get this in

parumpum committed:
Cleanup unused imports

Kuchenpirat committed:
Merge branch 'mealie-next' into fix/tag-filter

Kuchenpirat reviewed (changes requested):
When doing my final testing i found that this is still not working on the RecipeCardMobile. 
Probably an easy fix since it should just provide the same functionality as the normal RecipeCard.

parumpum committed:
Mobile recipe card fix

parumpum commented:
> When doing my final testing i found that this is still not working on the RecipeCardMobile. Probably an easy fix since it should just provide the same functionality as the normal RecipeCard.

Definitely was an easy fix thankfully!

Kuchenpirat committed:
Merge branch 'mealie-next' into fix/tag-filter

Kuchenpirat reviewed (approved):
Perfect, all good to go. Thanks for your work on this!
